Sort the result of file system traversal to avoid test failure on different
file systems.

Taken from upstream:
https://github.com/davidhalter/jedi/commit/7ff76bb7d0d94514e17cd1647f4ffaf533dd55f5

diff --git a/test/test_api/test_project.py b/test/test_api/test_project.py
--- a/test/test_api/test_project.py
+++ b/test/test_api/test_project.py
@@ -135,7 +135,7 @@ def test_search(string, full_names, kwargs, skip_pre_python36):
         defs = project.complete_search(string, **kwargs)
     else:
         defs = project.search(string, **kwargs)
-    assert [('stub:' if d.is_stub() else '') + d.full_name for d in defs] == full_names
+    assert sorted([('stub:' if d.is_stub() else '') + d.full_name for d in defs]) == full_names
 
 
 @pytest.mark.parametrize(
